cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutor.java

```diff
@@ -186,6 +202,80 @@ private Process launchCassandraNodeTask(@NotNull final Marker taskIdMarker, @Not
         return processBuilder.start();
     }
 
+    private static void modifyCassandraEnvSh(Marker taskIdMarker, CassandraNodeRunTask cassandraNodeTask) throws IOException {
+        int jmxPort = 0;
+        for (TaskEnv.Entry entry : cassandraNodeTask.getTaskEnv().getVariablesList()) {
+            if ("JMX_PORT".equals(entry.getName())) {
+                jmxPort = Integer.parseInt(entry.getValue());
+                break;
+            }
+        }
+
+        if (jmxPort == 7199 || jmxPort == 0) {
+            // Don't modify, if there's nothing to do...
+            return;
+        }
+
+        LOGGER.info(taskIdMarker, "Building cassandra-env.sh");
+
+        // Unfortunately it is not possible to pass JMX_PORT as an environment variable to C* startup -
+        // it is explicitly set in cassandra-env.sh
+
+        File cassandraEnvSh = new File("apache-cassandra-" + cassandraNodeTask.getVersion() + "/conf/cassandra-env.sh");
+
+        LOGGER.info(taskIdMarker, "Reading cassandra-env.sh");
+        List<String> lines = Files.readLines(cassandraEnvSh, Charset.forName("UTF-8"));
+        for (int i = 0; i < lines.size(); i++) {
+            String line = lines.get(i);
+            if (line.startsWith("JMX_PORT="))
+                lines.set(i, "JMX_PORT=\"" + jmxPort + '"');
+        }
+        LOGGER.info(taskIdMarker, "Writing cassandra-env.sh");
+        try (PrintWriter pw = new PrintWriter(new FileWriter(cassandraEnvSh))) {
+            for (String line : lines)
+                pw.println(line);
+        }
+    }
+
+    private static void modifyCassandraYaml(Marker taskIdMarker, CassandraNodeRunTask cassandraNodeTask) throws IOException {
+        LOGGER.info(taskIdMarker, "Building cassandra.yaml");
+
+        File cassandraYaml = new File("apache-cassandra-" + cassandraNodeTask.getVersion() + "/conf/cassandra.yaml");
+
+        Yaml yaml = new Yaml();
+        Map<String, Object> yamlMap;
+        LOGGER.info(taskIdMarker, "Reading cassandra.yaml");
+        try (BufferedReader br = new BufferedReader(new FileReader(cassandraYaml))) {
+            yamlMap = (Map<String, Object>) yaml.load(br);
```

Comment:
Please add an annotation to this method to suppress the compiler warning for unchecked cast.